### PR TITLE
Add support for the RNG on STM32L0

### PR DESCRIPTION
**Pull request by andysan**
The STM32L072 and STM32L073 come with a hardware random number generator. Add the necessary clock configuration and device tree changes to support the device.

Tested on a b_l072z_lrwan1 board (STM32L072).

### CODE REVIEW

**Comment by galak** [comment]
looks reasonable to me, will let @erwango approve

File: drivers/clock_control/clock_stm32l0_l1.c [hunk 1]
```diff
@@ -38,7 +38,8 @@ void config_pll_init(LL_UTILS_PLLInitTypeDef *pllinit)
  */
 void config_enable_default_clocks(void)
 {
-#if defined(CONFIG_EXTI_STM32) || defined(CONFIG_USB_DC_STM32)
+#if defined(CONFIG_EXTI_STM32) || defined(CONFIG_USB_DC_STM32) || \
+	(CONFIG_SOC_SERIES_STM32L0X && ENTROPY_STM32_RNG)
```
Comment (erwango) [inline drivers/clock_control/clock_stm32l0_l1.c:42]: I don't think symbol `ENTROPY_STM32_RNG` worked (should be `CONFIG_ENTROPY_STM32_RNG`).
If your samples worked with this piece of code, then maybe it is not needed.

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
@andysan Thanks for this contribution, can you additionally update board doc ?

File: drivers/clock_control/clock_stm32l0_l1.c (hunk 1 above)
Comment (andysan) [reply drivers/clock_control/clock_stm32l0_l1.c:42]: Whoops. The clock is definitely needed, but it was enabled because `CONFIG_EXTI_STM32` was set.

**Comment by andysan** [comment]
> @andysan Thanks for this contribution, can you additionally update board doc ?

Sure. Updated in new version.

File: drivers/clock_control/clock_stm32l0_l1.c (hunk 1 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32l0_l1.c]: Indeed.
Though it still needs to be fixed!

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by andysan** [comment]
Sorry about the screw-up with the rebase. Seems like I applied the clock fixup to the wrong commit.

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**